### PR TITLE
Full cv for pipeline results

### PR DESCRIPTION
**Pull request by sven1103**
This PR adds registration support for the result outputs for the following pipelines:

- ampliseq
- atacseq
- bcellmagic
- chipseq
- eager
- hlatyping
- mag
- metaboigniter
- methylseq
- mhcquant
- qbic-pipelines/cellranger
- qbic-pipelines/rnadeseq
- rnaseq
- sarek
- scrnaseq
- smrnaseq

### CODE REVIEW

File: src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy [hunk 1]
```diff
@@ -51,24 +51,62 @@ class NfCoreResultRegistry implements Registry {
...
+        tmpMap.put("nf-core/hlatyping", QExperimentType.Q_WF_NGS_HLATYPING)
+        tmpMap.put("nf-core/mag", QExperimentType.Q_WF_NGS_SHOTGUN_METAGENOMICS_ANALYIS)
+        tmpMap.put("nf-core/methylseq", QExperimentType.Q_WF_NGS_METHYLATION_ANALYSIS)
+        tmpMap.put("nf-core/mhcquant", QExperimentType.Q_WF_LX_MHC_QUANTIFICATION)
+        tmpMap.put("nf-core/rnaseq", QExperimentType.Q_WF_NGS_RNA_SEQ)
         tmpMap.put("nf-core/sarek", QExperimentType.Q_WF_NGS_VARIANT_CALLING)
+        tmpMap.put("nf-core/scrnaseq", QExperimentType.Q_WF_NGS_SCELL_RNA_SEQ)
+        tmpMap.put("nf-core/smrnaseq", QExperimentType.Q_WF_NGS_SMALL_RNA_SEQ)
+        tmpMap.put("qbic-pipelines/cellranger", QExperimentType.Q_WF_NGS_SCELL_RNA_SEQ)
+        tmpMap.put("qbic-pipelines/rnadeseq", QExperimentType.Q_WF_NGS_RNA_EXPRESSION_ANALYSIS)
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy:67]: I think there are some types missing: Q_WF_NGS_16S_TAXONOMIC_PROFILING, Q_WF_MX_IDENTIFY_AND_QUANTIFY_RUN

File: src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy [hunk 2]
```diff
@@ -51,24 +51,62 @@ class NfCoreResultRegistry implements Registry {
...
+        tmpMap.put("nf-core/metaboigniter", QSampleType.Q_WF_MX_IDENTIFY_AND_QUANTIFY_RUN)
+        tmpMap.put("nf-core/methylseq", QSampleType.Q_WF_NGS_METHYLATION_ANALYSIS_RUN)
+        tmpMap.put("nf-core/mhcquant", QSampleType.Q_WF_LX_MHC_QUANTIFICATION_RUN)
+        tmpMap.put("nf-core/rnaseq", QSampleType.Q_WF_NGS_RNA_SEQ_RUN)
+        tmpMap.put("nf-core/sarek", QSampleType.Q_WF_NGS_VARIANT_CALLING_RUN)
+        tmpMap.put("nf-core/scrnaseq", QSampleType.Q_WF_NGS_SCELL_RNA_SEQ_RUN)
+        tmpMap.put("nf-core/smrnaseq", QSampleType.Q_WF_NGS_SMALL_RNA_SEQ_RUN)
+        tmpMap.put("qbic-pipelines/cellranger", QSampleType.Q_WF_NGS_SCELL_RNA_SEQ_RUN)
+        tmpMap.put("qbic-pipelines/rnadeseq", QSampleType.Q_WF_NGS_RNA_EXPRESSION_RUN)
+        PIPELINE_TO_SAMPLE_TYPE = Collections.unmodifiableMap(tmpMap)
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy:89]: also missing here: Q_WF_NGS_16S_TAXONOMIC_PROFILING

File: src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy [hunk 3]
```diff
@@ -51,24 +51,62 @@ class NfCoreResultRegistry implements Registry {
...
+        tmpMap.put("nf-core/hlatyping", QSampleType.Q_WF_NGS_HLATYPING_RUN)
+        tmpMap.put("nf-core/mag", QSampleType.Q_WF_NGS_SHOTGUN_METAGENOMICS_RUN)
+        tmpMap.put("nf-core/metaboigniter", QSampleType.Q_WF_MX_IDENTIFY_AND_QUANTIFY_RUN)
+        tmpMap.put("nf-core/methylseq", QSampleType.Q_WF_NGS_METHYLATION_ANALYSIS_RUN)
+        tmpMap.put("nf-core/mhcquant", QSampleType.Q_WF_LX_MHC_QUANTIFICATION_RUN)
+        tmpMap.put("nf-core/rnaseq", QSampleType.Q_WF_NGS_RNA_SEQ_RUN)
+        tmpMap.put("nf-core/sarek", QSampleType.Q_WF_NGS_VARIANT_CALLING_RUN)
+        tmpMap.put("nf-core/scrnaseq", QSampleType.Q_WF_NGS_SCELL_RNA_SEQ_RUN)
+        tmpMap.put("nf-core/smrnaseq", QSampleType.Q_WF_NGS_SMALL_RNA_SEQ_RUN)
+        tmpMap.put("qbic-pipelines/cellranger", QSampleType.Q_WF_NGS_SCELL_RNA_SEQ_RUN)
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy:87]: Was it intended that this sample type is used multiple times?

Comment (jenniferboedker) [reply src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy:87]: Q_WF_NGS_SCELL_RNA_SEQ_RUN

**Comment by jenniferboedker** [comment]
[Add full mapping of nf-core pipelines to analysis types using CV](https://trello.com/c/1aGsb9Y3/810-add-full-mapping-of-nf-core-pipelines-to-analysis-types-using-cv)

File: src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy (hunk 3 above)
Comment (sven1103) [reply src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy:89]: yup, there are 2 pipelines that can do that type of analysis

File: src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy (hunk 1 above)
Comment (sven1103) [reply src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy]: nice spot!

File: src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy (hunk 2 above)
Comment (sven1103) [reply src/main/groovy/life/qbic/registration/handler/registries/NfCoreResultRegistry.groovy]: solved in a8bbf3e

File: src/main/groovy/life/qbic/registration/types/QExperimentType.groovy [hunk 4]
```diff
@@ -6,8 +6,20 @@ package life.qbic.registration.types
...
+    Q_WF_NGS_ATAC_SEQ,
+    Q_WF_NGS_ANCIENT_DNA_SEQ,
+    Q_WF_NGS_BCELL_REPERTOIRE_SEQ,
+    Q_WF_NGS_CHIP_SEQ,
+    Q_WF_NGS_HLATYPING,
+    Q_WF_NGS_METHYLATION_ANALYSIS,
     Q_WF_NGS_RNA_EXPRESSION_ANALYSIS,
+    Q_WF_NGS_RNA_SEQ,
+    Q_WF_NGS_SCELL_RNA_SEQ,
+    Q_WF_NGS_SHOTGUN_METAGENOMICS_ANALYIS,
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/registration/types/QExperimentType.groovy:21]:
```suggestion
    Q_WF_NGS_SHOTGUN_METAGENOMICS,
```

**Review by jenniferboedker — APPROVED** [review]

**Pull request merged**